vitals/BMSChecker.java

```diff
@@ -0,0 +1,39 @@
+package vitals;
+
+import java.util.function.Function;
+import static vitals.MonitorStates.*;
+
+public interface BMSChecker extends Function<BMS, Boolean> {
+	
+	static BMSChecker checkTemp(float temp) {
+		return bms -> checkInRange(TEMP,temp, bms.getTempLowerLimit(), bms.getTempUpperLimit());
+	}
+	
+	static BMSChecker checkSOC(float soc) {
+		return bms -> checkInRange(SOC,soc, bms.getSocLowerLimit(), bms.getSocUpperLimit());
+	}
+	
+	static BMSChecker checkChargeRate(float cr) {
+		return bms -> checkInRange(CR,cr, cr, bms.getChargeRateUpperLimit());
+	}
+	
+	static void printFailure(String text) {
+		System.out.println(text);
+	}
+	
+	static Boolean checkInRange(MonitorStates state, float value,float low, float high) {
```

Comment:
checkInRange Coupled with printFailure . Test environment expects printfailure() to be present while tetsing checkInRange